worlds/mlss/StateLogic.py

```diff
@@ -0,0 +1,100 @@
+def canDig(state, player):
+    return state.has("Green Goblet", player) and state.has("Hammers", player)
+
+
+def canMini(state, player):
+    return state.has("Red Goblet", player) and state.has("Hammers", player)
+
+
+def canDash(state, player):
+    return state.has("Red Pearl Bean", player) and state.has("Firebrand", player)
+
+
+def canCrash(state, player):
+    return state.has("Green Pearl Bean", player) and state.has("Thunderhand", player)
+
+
+def hammers(state, player):
+    return state.has("Hammers", player)
+
+
+def super(state, player):
+    return state.has("Hammers", player, 2)
+
+
+def ultra(state, player):
+    return state.has("Hammers", player, 3)
+
+
+def fruits(state, player):
+    return state.has("Red Chuckola Fruit", player) and state.has("Purple Chuckola Fruit", player) and state.has(
+        "White Chuckola Fruit", player)
+
+
+def pieces(state, player):
+    return state.has("Beanstar Piece 1", player) and state.has("Beanstar Piece 2", player) and state.has(
+        "Beanstar Piece 3", player) and state.has("Beanstar Piece 4", player)
+
+
+def neon(state, player):
+    return state.has("Blue Neon Egg", player) and state.has("Red Neon Egg", player) and state.has("Green Neon Egg",
+                                                                                                  player) and state.has(
+        "Yellow Neon Egg", player) and state.has("Purple Neon Egg", player) and state.has("Orange Neon Egg",
+                                                                                          player) and state.has(
+        "Azure Neon Egg", player)
```

Comment:
Interesting indents :D At this point I would just

```
def neon(state, player):
    return (
        state.has("Blue Neon Egg", player)
        and state.has("Red Neon Egg", player)
        and state.has("Green Neon Egg", player)
        and state.has("Yellow Neon Egg", player)
        and state.has("Purple Neon Egg", player)
        and state.has("Orange Neon Egg", player)
        and state.has("Azure Neon Egg", player)
    )
```

Incidentally, this is also what `ruff format worlds/mlss` does.